> Windows CI fails with `LINK : fatal error LNK1181: cannot open input file 'None.obj'`, when compiling the extension in `test_peg_generator`.

Yeah, no idea where that is coming from :(

--- a/Lib/test/test_peg_generator/test_c_parser.py
+++ b/Lib/test/test_peg_generator/test_c_parser.py
@@ -8,6 +8,7 @@ import sys
 
 from test import test_tools
 from test.test_peg_generator.ast_dump import ast_dump
+from test import support
 from pathlib import PurePath, Path
 from typing import Sequence
 
@@ -23,6 +24,9 @@ with test_tools.imports_under_tool('peg_generator'):
 
 class TestCParser(unittest.TestCase):
     def setUp(self):
+        cmd = support.missing_compiler_executable()
+        if cmd is not None:
+            self.skipTest('The %r command is not found' % cmd)
         self.tmp_path = tempfile.mkdtemp()
 
     def tearDown(self):

--- a/Tools/peg_generator/pegen/build.py
+++ b/Tools/peg_generator/pegen/build.py
@@ -25,13 +25,15 @@ MOD_DIR = pathlib.Path(__file__).parent
 def get_extra_flags(compiler_flags, compiler_py_flags_nodist):
     flags = sysconfig.get_config_var(compiler_flags)
     py_flags_nodist = sysconfig.get_config_var(compiler_py_flags_nodist)
+    if flags is None or py_flags_nodist is None:
+        return []
     return f'{flags} {py_flags_nodist}'.split()
 
 
 def compile_c_extension(
     generated_source_path: str,
     build_dir: Optional[str] = None,
-    verbose: bool = True,
+    verbose: bool = False,
     keep_asserts: bool = True,
 ) -> str:
     """Compile the generated source for a parser generator into an extension module.
@@ -47,12 +49,10 @@ def compile_c_extension(
     if verbose:
         distutils.log.set_verbosity(distutils.log.DEBUG)
 
-
     source_file_path = pathlib.Path(generated_source_path)
     extension_name = source_file_path.stem
     extra_compile_args = get_extra_flags('CFLAGS', 'PY_CFLAGS_NODIST')
     extra_link_args = get_extra_flags('LDFLAGS', 'PY_LDFLAGS_NODIST')
-    print(extra_compile_args, extra_link_args)
     if keep_asserts:
         extra_compile_args.append("-UNDEBUG")
     extension = [

--- a/Tools/peg_generator/pegen/build.py
+++ b/Tools/peg_generator/pegen/build.py
@@ -31,7 +31,7 @@ def get_extra_flags(compiler_flags, compiler_py_flags_nodist):
 def compile_c_extension(
     generated_source_path: str,
     build_dir: Optional[str] = None,
-    verbose: bool = False,
+    verbose: bool = True,
     keep_asserts: bool = True,
 ) -> str:
     """Compile the generated source for a parser generator into an extension module.
@@ -52,6 +52,7 @@ def compile_c_extension(
     extension_name = source_file_path.stem
     extra_compile_args = get_extra_flags('CFLAGS', 'PY_CFLAGS_NODIST')
     extra_link_args = get_extra_flags('LDFLAGS', 'PY_LDFLAGS_NODIST')
+    print(extra_compile_args, extra_link_args)
     if keep_asserts:
         extra_compile_args.append("-UNDEBUG")
     extension = [

